### ISSUE TITLE
`Optional[str]` triggers `int_parsing` instead of string parsing when used for `odmantic.Model`

### ISSUE THREAD

**Issue by Jibola** [art049/odmantic#379]
# Bug

When using `Optional[str]` in my class field definition, class instantiation fails because my Odmantic Model falsely expects an `int` type.

### Current Behavior

```
from typing import Optional

from odmantic import Model, Field
from pydantic import EmailStr

class Person(Model):
  hashed_password: Optional[str]
  totp_secret: Optional[str] = Field(default=None)
  totp_counter: Optional[int] = Field(default=None)
  

user = {
  "hashed_password": "hashed_password",
}
Person(**user)

> Traceback (most recent call last):
  File "/home/runner/WobblyScrawnyMegahertz/main.py", line 15, in <module>
    Person(**user)
  File "/home/runner/WobblyScrawnyMegahertz/.pythonlibs/lib/python3.10/site-packages/odmantic/model.py", line 544, in __init__
    super().__init__(**data)
  File "/home/runner/WobblyScrawnyMegahertz/.pythonlibs/lib/python3.10/site-packages/pydantic/main.py", line 164, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
pydantic_core._pydantic_core.ValidationError: 1 validation error for Person
hashed_password
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='hashed_password', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/int_parsing
```
### Expected behavior

```
...
Person(**user)
> Person(hashed_password='hashed_password', totp_secret=None, totp_counter=None)
```

### Environment

- ODMantic version: v1.0.0
- MongoDB version: v4.6.1
- Pydantic infos (output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())`):

```
             pydantic version: 2.5.2
        pydantic-core version: 2.14.5
          pydantic-core build: profile=release pgo=true
                 install path: /home/runner/WobblyScrawnyMegahertz/.pythonlibs/lib/python3.10/site-packages/pydantic
               python version: 3.10.11 (main, Apr  4 2023, 22:10:32) [GCC 12.2.0]
                     platform: Linux-6.2.0-1019-gcp-x86_64-with-glibc2.37
             related packages: typing_extensions-4.9.0 email-validator-2.1.0.post1

```

**Comment by art049** [comment]
Might be related to https://github.com/pydantic/pydantic/issues/8362

**Comment by anentropic** [comment]
oh wow...  I had this with an `Optional[datetime]` field and thought it was probably because PyMongo already fetched mongo `Date` as Python `datetime` and then Pydantic was expecting to deserialize JSON. I was wondering if maybe odmantic can only read data that it wrote itself (i.e. if it was inserting dates as strings to the db)

but it seems like it's actually just this bug, as I also had a couple of `Optional[str]` fields on the model... and either removing them or making my date field non-optional allows to fetch a row successfully (but then the model is no longer usable for my app)

this fix here https://github.com/art049/odmantic/pull/416/ works for me 👍

**Comment by art049** [comment]
Yes, this is fixed by #416. Amazing job @netomi!
